src/sql/src/partition.rs

```diff
@@ -38,117 +40,105 @@ macro_rules! between_string {
+    Partitions {
+        column_list: vec![Ident::new(ident)],
+        exprs: partition_rules_for_uuid(DEFAULT_PARTITION_NUM_FOR_TRACES, ident),
+    }
 }
 
-pub fn partition_rule_for_hexstring(ident: &str) -> Partitions {
-    let ident = Ident::new(ident);
-    let ident_expr = Expr::Identifier(ident.clone());
-
-    // rules are like:
-    //
-    // "trace_id < '1'",
-    // "trace_id >= '1' AND trace_id < '2'",
-    // "trace_id >= '2' AND trace_id < '3'",
-    // "trace_id >= '3' AND trace_id < '4'",
-    // "trace_id >= '4' AND trace_id < '5'",
-    // "trace_id >= '5' AND trace_id < '6'",
-    // "trace_id >= '6' AND trace_id < '7'",
-    // "trace_id >= '7' AND trace_id < '8'",
-    // "trace_id >= '8' AND trace_id < '9'",
-    // "trace_id >= '9' AND trace_id < 'A'",
-    // "trace_id >= 'A' AND trace_id < 'B' OR trace_id >= 'a' AND trace_id < 'b'",
-    // "trace_id >= 'B' AND trace_id < 'C' OR trace_id >= 'b' AND trace_id < 'c'",
-    // "trace_id >= 'C' AND trace_id < 'D' OR trace_id >= 'c' AND trace_id < 'd'",
-    // "trace_id >= 'D' AND trace_id < 'E' OR trace_id >= 'd' AND trace_id < 'e'",
-    // "trace_id >= 'E' AND trace_id < 'F' OR trace_id >= 'e' AND trace_id < 'f'",
-    // "trace_id >= 'F' AND trace_id < 'a' OR trace_id >= 'f'",
-    let rules = vec![
-        Expr::BinaryOp {
-            left: Box::new(ident_expr.clone()),
-            op: BinaryOperator::Lt,
-            right: Box::new(Expr::Value(Value::SingleQuotedString("1".to_string()))),
-        },
-        // [left, right)
-        between_string!(ident_expr, "1", "2"),
-        between_string!(ident_expr, "2", "3"),
-        between_string!(ident_expr, "3", "4"),
-        between_string!(ident_expr, "4", "5"),
-        between_string!(ident_expr, "5", "6"),
-        between_string!(ident_expr, "6", "7"),
-        between_string!(ident_expr, "7", "8"),
-        between_string!(ident_expr, "8", "9"),
-        between_string!(ident_expr, "9", "A"),
-        or!(
-            between_string!(ident_expr, "A", "B"),
-            between_string!(ident_expr, "a", "b")
-        ),
-        or!(
-            between_string!(ident_expr, "B", "C"),
-            between_string!(ident_expr, "b", "c")
-        ),
-        or!(
-            between_string!(ident_expr, "C", "D"),
-            between_string!(ident_expr, "c", "d")
-        ),
-        or!(
-            between_string!(ident_expr, "D", "E"),
-            between_string!(ident_expr, "d", "e")
-        ),
-        or!(
-            between_string!(ident_expr, "E", "F"),
-            between_string!(ident_expr, "e", "f")
-        ),
-        or!(
-            between_string!(ident_expr, "F", "a"),
-            Expr::BinaryOp {
+// partition_rules_for_uuid can creates partition rules up to 256 partitions.
+fn partition_rules_for_uuid(partition_num: u32, ident: &str) -> Vec<Expr> {
+    let ident_expr = Expr::Identifier(Ident::new(ident).clone());
+
+    let total_partitions = 256;
```

Comment:
I think we can go one step further by removing limit cap of 256. We can find the smallest total partition number of 16*n and generate the boundaries.

Reply:
Maybe up to 65536? For example:

```
[2,16]: use the first one hex char to partition;
[16, 256]: use the first two hex chars to partition;
...
[4097, 65536]
```

Maybe it's enough for a very, very long time.

Reply:
Also, I add a constraint: the partition number should be the order of `2`(https://github.com/GreptimeTeam/greptimedb/pull/5743/files#diff-f01f6e55255c1b45965db4f395a14ff3d1b7a50a0c1d8b8163b88e066ee1f1caR59).

Then we can create 2,4,8,16,32,...,65536 partitions by this API.